editor/editor_help_search.cpp

```diff
@@ -124,7 +145,7 @@ void EditorHelpSearch::_notification(int p_what) {
 		case NOTIFICATION_PROCESS: {
 			// Update background search.
 			if (search.is_valid()) {
-				if (search->work()) {
+				if (search->work(1000)) {
```

Comment:
Worth noting that this seems to be the only call for `work()`, so if overriding the default value for its parameter, I question whether the parameter should have a default value in the first place (or if it should be overridden there too).

But either way we might need a comment to justify the magic number. IIUC this reduces the time allowed on each iteration to 1 ms - while the typical framerate with VSync should be 60 FPS so 16.6 ms. Isn't this significantly slowing down the search, forcing it to run over more `_process` iterations than it needs to?
The previous default of 100000 µs = 100 ms was likely too much indeed though, as it's the time for an average of 6 frames. Maybe this should be 10000 instead?

Reply:
Thank you, I am in the process of tweaking this at the moment, will test this suggestion

Reply:
I tested this PR both as is (1000) and with 10000 instead and the latter feels much faster (in a `target=debug` build on Linux). But the original 100000 is even better, there's no visible delay.

1000 (this PR):
https://user-images.githubusercontent.com/4701338/176634518-deb25591-3152-4d55-8ff7-dbc4dacffa0c.mp4

10000 (my above suggestion, though it can likely be refined further):
https://user-images.githubusercontent.com/4701338/176634612-fc18a382-6a98-45d1-8cff-d00f69811505.mp4

100000 (original):
https://user-images.githubusercontent.com/4701338/176635148-f09c48eb-108b-4d3a-b8e4-a5d8b4cf43db.mp4

Reply:
Thank you! The delay is specifically when search is length < 2 as there are a lot of entries

Reply:
Here's an example with a longer string:

1000 (still visible delay and flicker):

https://user-images.githubusercontent.com/4701338/176636397-06211660-88a8-4f49-80fd-06794fc12446.mp4

100000 (no delay, no flicker):

https://user-images.githubusercontent.com/4701338/176636506-34040453-7031-4c99-bb61-0241471ed691.mp4

So it seems to me that the original value was pretty good. Why did you need to change it?
Does this related to the crash fix you mention in the description?

Reply:
What are your results like for short searches? I'm looking at changing the behavior between short and long searches and will amend in a moment.

Reply:
The crash is not directly related but resolved by clearing the runner when popup is hidden, as when searches are long the runner will keep running after the tree has been cleared causing it to crash when trying to add children to non-existing tree items

Reply:
Added two modes, one for short 0-1 characters, and one for longer "proper" searches, see how they feel

Reply:
> What are your results like for short searches length < 2? I'm looking at changing the behavior between short and long searches and will amend in a moment.

Indeed for searches with length < 2 it makes a big difference, 1000 faster than 100000, 10000 seems maybe fastest?

1000:

https://user-images.githubusercontent.com/4701338/176639590-3c380462-0d3c-43be-b51e-ea1c830def50.mp4

10000:

https://user-images.githubusercontent.com/4701338/176639607-15b38332-4866-4fb3-84f9-3d86d420882c.mp4

100000:

https://user-images.githubusercontent.com/4701338/176639614-fe383cdd-0877-42e1-b34d-dc8430c8b2f9.mp4

But I think trying to optimize for searches with length < 2 which are practically not searches but showing the whole index is the wrong approach. The content should never need to change for such searches, so it can simply be cached the first time the EditorHelp is used, and reused whenever the search string goes below 2. Then you don't even need to add logic to handle checking if going from 0 to 1 or 1 to 0, it would likely be fast enough to just refresh the tree with precomputed data so it doesn't matter (to be confirmed).

Reply:
Thank you, that is a great idea, I'll see how I can resolve that later today, thank you for your help!